internal/server/handlers/handler.go

```diff
@@ -94,3 +95,12 @@ type MonitoringInstanceHandler interface {
 	GetMonitoringInstance(ctx context.Context, namespace, name string) (*everestv1alpha1.MonitoringConfig, error)
 	DeleteMonitoringInstance(ctx context.Context, namespace, name string) error
 }
+
+// PodSchedulingPolicyHandler provides methods for handling operations on pod scheduling policies.
+type PodSchedulingPolicyHandler interface {
+	CreatePodSchedulingPolicy(ctx context.Context, psp *everestv1alpha1.PodSchedulingPolicy) (*everestv1alpha1.PodSchedulingPolicy, error)
+	UpdatePodSchedulingPolicy(ctx context.Context, name string, psp *everestv1alpha1.PodSchedulingPolicy) (*everestv1alpha1.PodSchedulingPolicy, error)
```

Comment:
why do we need the name as a separate parameter? Only the `rbacHandler` needs it (`validateHandler` and `k8sHandler` does not need it), so the `rbacHandler` could extract the name from the `PodSchedulingPolicy` and we could avoid passing this mostly unused parameter through the whole chain.